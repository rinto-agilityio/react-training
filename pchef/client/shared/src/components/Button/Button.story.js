// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Helper
import { wInfo } from '../../../.storybook/utils';

// Component
import Button from './index'

storiesOf('Button', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Button
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Success', () => (
    <Button
      type="success"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Danger', () => (
    <Button
      type="danger"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Warning', () => (
    <Button
      type="warning"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Info', () => (
    <Button
      type="info"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))
