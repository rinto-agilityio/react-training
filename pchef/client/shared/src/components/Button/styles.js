// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

// Create style for button
const styles = StyleSheet.create({
  button: {
    ...METRICS.flexCenter,
    padding: METRICS.padding.sm,
  },
  title: {
    color: COLORS.white,
    textTransform: 'uppercase',
    fontWeight: FONTS.fontWeight.large,
  },
  default: {
    backgroundColor: COLORS.baseBlue,
  },
  success: {
    backgroundColor: COLORS.green,
  },
  danger: {
    backgroundColor: COLORS.red,
  },
  warning: {
    backgroundColor: COLORS.yellow,
  },
  info: {
    backgroundColor: COLORS.darkWhite,
  },
})

export default styles
