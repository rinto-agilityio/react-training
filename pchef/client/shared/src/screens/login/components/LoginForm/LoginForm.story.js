// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { View } from 'react-native'
import { action } from '@storybook/addon-actions'

// Components
import LoginForm from '.'

// define props
const props = {
  onSubmit: action('clicked'),
  setError: action('clicked'),
}

storiesOf('LoginForm', module)
  .add('default', () => (
    <View>
      <LoginForm {...props} />
    </View>
  ))
  .add('error', () => (
    <View>
      <LoginForm {...props} type="primary" error />
    </View>
  ))
