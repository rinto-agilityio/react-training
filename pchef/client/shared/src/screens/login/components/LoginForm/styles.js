// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS, COLORS, FONTS } from '../../../../themes'

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.white,
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  primaryContainer: {
    width: METRICS.largeWrapper,
    paddingTop: METRICS.largePadding * 2,
    paddingRight: METRICS.extraLargePadding,
    paddingBottom: METRICS.smallPadding + METRICS.largePadding,
    paddingLeft: METRICS.extraLargePadding,
    marginTop: METRICS.largeMargin * 2,
  },
  secondaryContainer: {
    width: METRICS.mediumWrapper * 2,
    paddingTop: METRICS.extraLargePadding * 2,
    paddingRight: METRICS.extraLargePadding + METRICS.largePadding,
    paddingBottom: METRICS.largePadding * 2 + METRICS.smallPadding,
    paddingLeft: METRICS.extraLargePadding + METRICS.largePadding,
  },
  formGroup: {
    borderRadius: METRICS.baseBorderRadius,
    marginBottom: METRICS.smallMargin + METRICS.mediumMargin,
    borderColor: COLORS.ghost,
    borderWidth: METRICS.smallBorderWidth,
    width: '100%',
  },
  errorFormGroup: {
    borderColor: COLORS.baseRed,
  },
  input: {
    borderRadius: METRICS.baseBorderRadius,
    borderBottomWidth: 0,
    paddingLeft: METRICS.smallPadding + METRICS.mediumPadding,
    paddingRight: METRICS.smallPadding + METRICS.mediumPadding,
  },
  primaryInput: {
    fontSize: FONTS.fontSize.base,
  },
  buttonWrapper: {
    borderRadius: METRICS.baseBorderRadius,
    padding: 0,
  },
  button: {
    height: METRICS.mediumHeight,
    borderRadius: METRICS.baseBorderRadius,
    backgroundColor: COLORS.mantisGreen,
  },
  primaryButton: {
    width: METRICS.largeButton,
  },
  secondaryButton: {
    width: METRICS.largeWrapper,
  },
})

export default styles
