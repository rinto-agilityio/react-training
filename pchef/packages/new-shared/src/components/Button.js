/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';
import { Provider as PaperProvider, Button as PaperButton } from 'react-native-paper';

type Props = {};
export default class CustomButton extends Component<Props> {
  render() {
    return (
      <PaperProvider>
        <View style={styles.container}>
          <PaperButton
            mode="contained"
            onPress={() => console.log('PaperButton')}
          >
            PaperButton
          </PaperButton>
        </View>
      </PaperProvider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});
