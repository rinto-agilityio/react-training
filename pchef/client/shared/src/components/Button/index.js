// libs
import React from 'react'
import { Button as ButtonComponent, Text } from 'react-native-paper'

// Styles
import styles from './styles'
import { COLORS } from '../../themes/index'

type Props = {
  onPress?: () => void,
  buttonStyle?: {} | Array<{}>,
  title: string,
  titleStyle?: {} | Array<{}>,
  disabled?: boolean,
  mode?: string,
  contentStyle?: {} | Array<{}>,
  color?: string,
  loading?: boolean,
  type?: string,
}

const Button = ({
  onPress,
  buttonStyle,
  title,
  mode,
  type,
  disabled,
  contentStyle,
  color,
  loading,
  titleStyle,
}: Props) => (
  <ButtonComponent
    color={color}
    loading={loading}
    onPress={onPress}
    mode={mode}
    disabled={disabled}
    style={[styles.button, styles[type], buttonStyle]}
    contentStyle={[styles.content, contentStyle]}
  >
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </ButtonComponent>
)

Button.defaultProps = {
  onPress: () => {},
  buttonStyle: {},
  titleStyle: {},
  disabled: false,
  mode: 'contained',
  contentStyle: {},
  color: COLORS.baseBlue,
  loading: false,
  type: 'default',
}

export default Button
