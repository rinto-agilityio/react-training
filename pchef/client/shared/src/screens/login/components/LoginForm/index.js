// Libs
import React, { useRef } from 'react'
import { View } from 'react-native'

// Components
import TextBox from '../../../../components/TextBox'
import Button from '../../../../components/Button'
import Wrapper from '../../../../layout/Wrapper'
import { PATTERN } from '../../../../constants'

// Styles
import styles from './styles'

type Props = {
  customStyles?: Object,
  onSubmit: (email: string, password: string) => Promise<void>,
  type?: string,
  setError: (error: boolean) => void,
  error?: boolean,
  isSubmit?: boolean,
}

const LoginForm = ({
  customStyles = {},
  onSubmit,
  type = 'primary',
  setError,
  error,
  isSubmit,
}: Props) => {
  // define user email ref and password ref
  const emailRef = useRef(null)
  const passwordRef = useRef(null)

  // handling submit form
  const handlingSubmit = () => {
    // get email input and password input
    const email = emailRef.current ? emailRef.current._node.value.trim() : ''
    const password = passwordRef.current
      ? passwordRef.current._node.value.trim()
      : ''

    // validate email and password
    if (PATTERN.EMAIL.test(email) && password.length >= 8) {
      setError(false)
      onSubmit(email, password)
    } else {
      setError(true)
    }
  }

  // input style
  const inputStyles = [styles.input, styles[`${type}Input`]]

  // form group style
  const formGroupStyle = [
    styles.formGroup,
    styles[`${type}FormGroup`],
    error ? styles.errorFormGroup : null,
  ]

  return (
    <Wrapper
      direction="column"
      customStyles={[
        styles.container,
        styles[`${type}Container`],
        customStyles,
      ]}
    >
      <View style={formGroupStyle}>
        <TextBox
          placeholder="username@asnet.com.vn"
          placeholderTextColor="darkgray"
          customStyle={inputStyles}
          refInput={emailRef}
        />
      </View>
      <View style={formGroupStyle}>
        <TextBox
          placeholder="Password at least 8 characters"
          placeholderTextColor="darkgray"
          customStyle={inputStyles}
          refInput={passwordRef}
          secureTextEntry
        />
      </View>
      <View style={[styles.formGroup, styles[`${type}FormGroup`]]}>
        <Button
          title="Login"
          contentStyle={[styles.button, styles[`${type}Button`]]}
          buttonStyle={styles.buttonWrapper}
          onPress={handlingSubmit}
          disabled={isSubmit}
        />
      </View>
    </Wrapper>
  )
}

LoginForm.defaultProps = {
  customStyles: {},
  type: 'primary',
  error: false,
  isSubmit: false,
}

export default LoginForm
