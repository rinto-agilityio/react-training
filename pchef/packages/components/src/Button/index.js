// libs
import React from 'react';
import { TouchableOpacity, StyleSheet, Text } from 'react-native';

// Styles
import styles from './styles';

const Button = ({
  onClick,
  customStyle,
  size,
  title,
}) => (
  <TouchableOpacity
    onPress={onClick}
    style={[
      customStyle,
      styles.button,
      styles[size]
    ]}
  >
    <Text
      style={styles.title}
    >
      {title}
    </Text>
  </TouchableOpacity>
)

export default Button

Button.defaultProps = {
  onClick: () => { },
  customStyle: {},
  size: 'default',
}
