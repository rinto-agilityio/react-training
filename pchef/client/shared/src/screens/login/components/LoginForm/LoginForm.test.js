// Components
import LoginForm from '.'
import Button from '../../../../components/Button'

it('LoginForm snapshots', () => {
  // mock props
  const props = {
    onSubmit: jest.fn(),
    setError: jest.fn(),
    error: false,
    isSubmit: false,
  }

  const primaryComponent = renderer.create(
    <LoginForm {...props} type="primary" customStyle={{ flex: 1 }} />
  )
  const secondaryComponent = renderer.create(
    <LoginForm {...props} error isSubmit type="secondary" />
  )

  expect(primaryComponent).toMatchSnapshot()
  expect(secondaryComponent).toMatchSnapshot()
})

it('LoginForm actions', () => {
  // mock props
  const props = {
    onSubmit: jest.fn(),
    setError: jest.fn(),
  }

  const component = shallow(<LoginForm {...props} />)

  component
    .find(Button)
    .at(0)
    .props()
    .onPress()
  expect(props.setError).toHaveBeenCalled()
})
