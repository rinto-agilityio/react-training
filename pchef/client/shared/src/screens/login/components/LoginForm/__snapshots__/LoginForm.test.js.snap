// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshots 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-14lw9ot r-display-6koalj r-flexDirection-eqz5dr r-flexWrap-1w6e6rj r-justifyContent-a2tzq0 r-marginLeft-1jj8364 r-marginRight-lchren r-marginTop-2o02ov r-paddingBottom-b6sd5j r-paddingLeft-1vwolk8 r-paddingRight-rnv2vh r-paddingTop-1ur9v65 r-width-1ipicw7"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "WebkitFlexGrow": 0,
      "flexGrow": 0,
      "msFlexPositive": 0,
    }
  }
>
  <div
    className="css-view-1dbjc4n r-borderColor-1qh06s0 r-borderRadius-1xfd6ze r-borderWidth-rs99b7 r-marginBottom-ku1wi2"
  >
    <div
      className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz"
    >
      <input
        autoCapitalize="none"
        autoComplete="on"
        autoCorrect="off"
        className="css-textinput-1cwyjr8 r-borderBottomWidth-ndvcnb r-borderColor-34efmp r-borderRadius-1xfd6ze r-color-cqee49 r-flex-13awgt0 r-fontSize-ubezar r-height-eu3ka r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-placeholderTextColor-d9raxc"
        data-focusable={true}
        defaultValue=""
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onSelect={[Function]}
        placeholder="username@asnet.com.vn"
        readOnly={false}
        spellCheck={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-borderColor-1qh06s0 r-borderRadius-1xfd6ze r-borderWidth-rs99b7 r-marginBottom-ku1wi2"
  >
    <div
      className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz"
    >
      <input
        autoCapitalize="none"
        autoComplete="on"
        autoCorrect="off"
        className="css-textinput-1cwyjr8 r-borderBottomWidth-ndvcnb r-borderColor-34efmp r-borderRadius-1xfd6ze r-color-cqee49 r-flex-13awgt0 r-fontSize-ubezar r-height-eu3ka r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-placeholderTextColor-d9raxc"
        data-focusable={true}
        defaultValue=""
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onSelect={[Function]}
        placeholder="Password at least 8 characters"
        readOnly={false}
        spellCheck={false}
        type="password"
      />
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-borderColor-1qh06s0 r-borderRadius-1xfd6ze r-borderWidth-rs99b7 r-marginBottom-ku1wi2"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-borderRadius-1xfd6ze r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(66,165,245,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "minWidth": "64px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1umu7ot r-borderRadius-1xfd6ze r-flexDirection-18u37iz r-height-1iud8zs r-justifyContent-1777fci r-width-8upyzv"
        >
          <div
            className="css-text-76zvg2 css-textOneLine-bfa6kz r-color-cqee49 r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-76zvg2 css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              Login
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LoginForm snapshots 2`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-14lw9ot r-display-6koalj r-flexDirection-eqz5dr r-flexWrap-1w6e6rj r-justifyContent-a2tzq0 r-marginLeft-1jj8364 r-marginRight-lchren r-paddingBottom-2o15sn r-paddingLeft-1fd96xs r-paddingRight-yongw r-paddingTop-u04ygo r-width-1rttkqs"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "WebkitFlexGrow": 0,
      "flexGrow": 0,
      "msFlexPositive": 0,
    }
  }
>
  <div
    className="css-view-1dbjc4n r-borderColor-4a18lf r-borderRadius-1xfd6ze r-borderWidth-rs99b7 r-marginBottom-ku1wi2"
  >
    <div
      className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz"
    >
      <input
        autoCapitalize="none"
        autoComplete="on"
        autoCorrect="off"
        className="css-textinput-1cwyjr8 r-borderBottomWidth-ndvcnb r-borderColor-34efmp r-borderRadius-1xfd6ze r-color-cqee49 r-flex-13awgt0 r-fontSize-1i10wst r-height-eu3ka r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-placeholderTextColor-d9raxc"
        data-focusable={true}
        defaultValue=""
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onSelect={[Function]}
        placeholder="username@asnet.com.vn"
        readOnly={false}
        spellCheck={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-borderColor-4a18lf r-borderRadius-1xfd6ze r-borderWidth-rs99b7 r-marginBottom-ku1wi2"
  >
    <div
      className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz"
    >
      <input
        autoCapitalize="none"
        autoComplete="on"
        autoCorrect="off"
        className="css-textinput-1cwyjr8 r-borderBottomWidth-ndvcnb r-borderColor-34efmp r-borderRadius-1xfd6ze r-color-cqee49 r-flex-13awgt0 r-fontSize-1i10wst r-height-eu3ka r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-placeholderTextColor-d9raxc"
        data-focusable={true}
        defaultValue=""
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onSelect={[Function]}
        placeholder="Password at least 8 characters"
        readOnly={false}
        spellCheck={false}
        type="password"
      />
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-borderColor-1qh06s0 r-borderRadius-1xfd6ze r-borderWidth-rs99b7 r-marginBottom-ku1wi2"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-borderRadius-1xfd6ze r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(66,165,245,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "minWidth": "64px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        aria-disabled={true}
        className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-userSelect-lrvibr"
        disabled={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1umu7ot r-borderRadius-1xfd6ze r-flexDirection-18u37iz r-height-1iud8zs r-justifyContent-1777fci r-width-1ipicw7"
        >
          <div
            className="css-text-76zvg2 css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,0.32)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-76zvg2 css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              Login
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
